controllers/provisioners/config_test.go

```diff
@@ -160,6 +169,333 @@ spec:
 	g.Expect(c.InstanceGroup.Spec.EKSSpec.EKSConfiguration.KeyPairName).To(gomega.Equal("TestKeyPair"))
 }
 
+func TestSetDefaultsWithRestrictedConditional(t *testing.T) {
+	var (
+		g = gomega.NewGomegaWithT(t)
+	)
+
+	// Restricted fields are always replaced with default values
+
+	mockBoundaries := `
+    restricted:
+    - spec.eks.configuration.keyPairName
+    - spec.eks.configuration.taints
+    - spec.eks.configuration.labels
+    - spec.eks.configuration.securityGroups
+    - spec.eks.configuration.instanceType
+    - spec.eks.configuration.image
+    - spec.strategy`
+
+	mockConditionals := `
+- annotation: "instancemgr.keikoproj.io/os-family=windows"
+  defaults:
+    spec:
+      eks:
+        configuration:
+          image: ami-22222222222
+- annotation: "instancemgr.keikoproj.io/arch=arm64"
```

Comment:
So if we have two annotations referencing the same default value, the last one always takes priority? how do we choose between the two?

Also should use the yaml content for defaults be inline since its unstructured?
```
- annotation: <annotation>
  defaults: |
    < inline yaml >
```

Reply:
Yeah, the priority is just the last applicable rule that applies. We could consider doing something around a `priority` field where folks can determine explicitly what the ordering is.

Reply:
w.r.t. the inline YAML - I don't think it's needed for defaults since the entire conditionals list is already defined like:
```
kind: ConfigMap
data: 
    conditionals: |
      - annotation: "x=y"
        defaults: 
          foo: bar
```

Reply:
Is ordering kept though when we unmarshal to yaml? or do we have to sort? priority might be good.. maybe instead we should allow chaining of annotations:

```
- annotation: "instancemgr.keikoproj.io/os-family=windows,instancemgr.keikoproj.io/arch=arm64"
  defaults:
    spec:
      eks:
        configuration:
          image: ami-22222222222
```

would this be a better mechanism?

Reply:
It seems to be preserved in the tests that I've run - and I'd expect it to remain ordered based on previous use - I think order issues tend to happen when folks want ordering on things that are not supposed to be ordered - typically k/v maps.

Reply:
Maybe we can keep the current ordering, but still allow having conditionals with multiple annotations? although if there are multiple the string might get pretty long.. but it might be useful since it will allow avoiding thinking of order when there are multiple conditions for a default

Reply:
Also might be worth looking into supporting standard selector syntax

Reply:
This actually might be nicer and allow the chaining, and will also avoid the string parsing / validation. we can add more matchers later
```
selector:
  matchAnnotations:
    instancemgr.keikoproj.io/os-family: windows
    instancemgr.keikoproj.io/arch=arm64
  defaults:
    spec:
      eks:
        configuration:
          image: ami-22222222222
```

Reply:
check out the latest commit - I ended up leveraging the K8S field selector library so we can do things like:

```yaml
- annotation: instancemgr.keikoproj.io/os-family = 'windows', instancemgr.keikoproj.io/arch: arm64
  defaults:
  spec:
    eks:
      configuration:
        image: ami-22222222222
- annotation: instancemgr.keikoproj.io/os-family in (bottlerocket, amazonlinux2)
  defaults:
  spec:
    eks:
      configuration:
        image: ami-333333333

```

Reply:
@backjo this is awesome!
I think the only downside, is that if someone wants to match multiple annotations it can become a super long string - but this seems super useful and we didn't have to implement any of our own parsing which is great.

Reply:
Wonder if we should change the key "annotation" to "annotationSelector" WDYT?

Reply:
yep, totally.